Implement  migration from Leap+textmode to SLES

hjluo opened the pull request:
Now we support leap15.2 textmode to SLE15SP2 migration through 'zypper migration' and with 
yast UI. We added the following 2 methods of migration. And migration should be leap 15.2 to sles15SP2 or leap15.3 to sles15SP3.

1. Add migration from Leap 15.2 textmode to SLES15SP2 with yast UI.
2. Add migration from Leap 15.2 textmode to SLES15SP2 with 'zypper migration'.

During the test, we added a needle bootmenu-LEAP-15-2-20200708 with tag grub2.

- Related ticket: 
         https://progress.opensuse.org/issues/52460
         https://progress.opensuse.org/issues/68609

- Needles: N/A
  
- Verification run: 
         http://openqa.nue.suse.com/tests/4459604 yast
         http://openqa.nue.suse.com/tests/4459605 zypper

lemon-suse commented:
Please update the title of the PR, it is not make sense now.

If you have added Needles and added through OSD, it is better list them in the description for that belongs the scope of the PR also.

lemon-suse commented on lib/registration.pm:
```diff
@@ -720,7 +720,14 @@ sub registration_bootloader_params {
+    # for leap to sle migration, we need to install yast2-registration
+    # before running yast2 registration module.
+    my $client_module = 'scc';
+    if (is_leap_migration) {
```
Not always need to install yast2-registration even it is for leap migration, such as zypper migration for leap to sles.

hjluo commented:
> Please update the title of the PR, it is not make sense now.
> 
> If you have added Needles and added through OSD, it is better list them in the description for that belongs the scope of the PR also.

Thanks, changed.
